tests/test_watson.py

```diff
@@ -195,6 +195,75 @@ def test_frames_with_empty_given_state(config_dir, mock):
     assert len(watson.frames) == 0
 
 
+def test_frames_filter(watson):
+    samples = (
+        ('foo', ('A')),
+        ('bar', ('A')),
+        ('foo', ('B')),
+        ('lol', ('B')),
+        ('bar', ('A'))
+    )
+    for name, tags in samples:
+        watson.frames.add(name, 4000, 4000, tags)
+
+    foo_projects = watson.frames.filter(projects=('foo'))
+    len = 0
+    for frame in foo_projects:
+        assert frame.project == 'foo'
+        len += 1
+    assert len == 2
```

Comment:
I would find the following more readable:
```
assert all(frame.project == 'foo' for frame in foo_projects)
assert len(foo_projects) == 2
```

This is true for all the other cases of course :) .